test/parachain/tests/integration_test.rs

```diff
@@ -329,21 +329,72 @@ async fn integration_test() {
 				"--bootnodes=/ip4/127.0.0.1/tcp/27116/p2p/{}",
 				polkadot_bob_id
 			))
+			.arg("--charlie")
 			.spawn()
 			.unwrap();
-		let cumulus_helper = ChildHelper::new("cumulus", &mut cumulus);
+		let cumulus_charlie_helper = ChildHelper::new("cumulus-charlie", &mut cumulus_charlie);
 		wait_for_tcp("127.0.0.1:27017").await;
 
 		// connect rpc client to cumulus
-		let transport_client_cumulus =
+		let transport_client_cumulus_charlie =
 			jsonrpsee::transport::http::HttpTransportClient::new("http://127.0.0.1:27017");
-		let mut client_cumulus = jsonrpsee::raw::RawClient::new(transport_client_cumulus);
+		let mut client_cumulus_charlie = jsonrpsee::raw::RawClient::new(transport_client_cumulus_charlie);
 
 		// wait for parachain blocks to be produced
 		let number_of_blocks = 4;
 		let mut previous_blocks = HashSet::with_capacity(number_of_blocks);
 		loop {
```

Comment:
I expect that these tests are split and not that we just grow this one test. Or that we make the tests based on each other (by reusing the db or similar).

However, we should at least start to refactor stuff into functions. You just copied the same code.

Reply:
I just refactored. Should I go further or is it good for now?